message ExpansionsRequest {
  optional string short = 1;
  optional string user = 2;
  optional string target = 3;
  optional string url = 4;  
  optional string start_date = 5;
  optional string end_date = 6;
}

message ExpansionsResponse {
  optional string short = 1;
  optional string user = 2;
  optional string url = 3;  
  optional string target = 4;

  message Expansion {
      optional string detection_date = 1;
      optional string ua_string = 2;
      optional string ua_name = 3;
      optional string ua_family = 4;
      optional string ua_company = 5;
      optional string ua_type = 6;
      optional string os_family = 7;
  }

  optional int32 page_number = 5;
  optional int32 results_per_page = 6;
  optional int32 page_count = 7;
  repeated Expansion expansion = 8;
}
